### PR TITLE
Changes for AmqpAnnotatedMessage to use BinaryData (only core-amqp changes)

### PR DESCRIPTION
**Pull request by hemanttanwar**
Taking work from this PR and putting it in a feature branch.

https://github.com/Azure/azure-sdk-for-java/pull/16895

We are not breaking API , This was decided after discussion with .Net and Jonathan. 

Note: But There are new discussion happening with Clemens as  we speak and they could bring new breaking changes.

This change is done on separate feature  branch so that we can use this for beta release. (Because IT is using  azure-core-experimental)
- I will have separate PR to move this on to master
     - By removing  azure-core-experimental dependency.

### CODE REVIEW

File: sdk/core/azure-core-amqp/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Release History
 
 ## 1.7.0-beta.1 (Unreleased)
+### New Features
+- Adding support for `BinaryData` in `AmqpAnnotatedMessage`.
+ 
+### Breaking Changes
+- Updated constructor `AmqpDataMessageBody` to support `BinaryData`.
+- Renamed `AmqpDataBody` to `AmqpDataMessageBody`.
```
Comment (hemanttanwar) [inline sdk/core/azure-core-amqp/CHANGELOG.md:9]: This is to match .net type name.

**Comment by joshfree** [comment]
I don't think we have something similar published (yet) in azure-sdk, but we should be following a compat bar similar to this: https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/breaking-changes.md#bucket-1-public-contract

E.g., we can't make breaking changes as defined in "buckets 1-3" to already GA'd libraries.

/cc @KrzysztofCwalina

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpBodyType.java [hunk 2]
```diff
@@ -4,19 +4,23 @@
...
+     * by AMQP.
      */
     VALUE,
+
     /**
-     * Represent Amqp Sequence type
+     * When message content is a single object, equivalent to AMQP Value. The object must be of a type supported by
+     * AMQP.
      */
     SEQUENCE;
```
Comment (JonathanGiles) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpBodyType.java:25]: Are these JavaDocs correctly associated with the right enum value? It seems to me you have them back to front.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpDataMessageBody.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * This represent amqp payload of {@link AmqpBodyType#DATA} type.
+ */
+public final class AmqpDataMessageBody implements AmqpMessageBody {
+    private final ClientLogger logger = new ClientLogger(AmqpDataMessageBody.class);
+    private final BinaryData binaryData;
+
+    /**
+     * Creates instance of {@link AmqpDataMessageBody} with given {@link Iterable} of {@link BinaryData}. Please note
+     * that this version of the SDK supports only one element in given {@link Iterable} of {@link BinaryData}.
```
Comment (JonathanGiles) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpDataMessageBody.java:22]: Why do we only support one value in an iterable?

**Review by JonathanGiles — COMMENTED** [review]
Before any further breaking changes are made, you need to be extremely confident that you are at parity with .NET and that no further breaks will be required at all.

File: sdk/core/azure-core-amqp/CHANGELOG.md (hunk 1 above)
Comment (srnagar) [reply sdk/core/azure-core-amqp/CHANGELOG.md:9]: This list should also include the breaking change to `userId`.

**Comment by hemanttanwar** [comment]
OLD PR closing it.I will have  amqp changes in another fresh PR.

**Pull request closed without merging**